### PR TITLE
Character animation spring cleaning and bug fixing

### PR DESCRIPTION
**Pull request by Raketfart**
This fixes a couple of bugs.

The character facing wasn't being updated, when encountering a door, and waiting for it to open.

The character inventory sortorder was sometimes flipping in front and behind the character.

Then there is some cleanup of the animations. 
The characterAnimation now uses the same "SpriteNameAnimation" as FurnitureAnimation. So the sprites are back in the hands of the SpriteManager, instead of being stored in the animation.

### CODE REVIEW

**Comment by koosemose** [comment]
I'm not entirely sure if this is caused by your code, or just made evident due to your updates. But at certain times (most often when leaving the top airlock door, and heading directly to the left), the character will end up facing odd ways, either when waiting on the door (several times facing to the right even though they came from the right and will be heading left), or when going through the door (appear to walk through the door facing to one side or another), and occasionally a tile or two after leaving the door (continue facing up, when moving directly to the left).

**Comment by Raketfart** [comment]
I thought i fixed that. I'll take a look again later. Thanks

**Comment by Raketfart** [comment]
Found the bug, and I think it means that this glitch could also have happened with furniture. 
Never again.

File: Assets/Scripts/Controllers/CharacterSpriteController.cs [hunk 1]
```diff
@@ -103,10 +102,13 @@ protected override void OnChanged(Character c)
...
         // Make sure the furniture's graphics are correct.
         SpriteRenderer inv_sr = objectGameObjectMap[c].transform.GetChild(0).gameObject.GetComponent<SpriteRenderer>();
+
+        // Important to set the characters SortOrder first.
+        int charSortOrder = c.animation.SetAndGetSortOrder();
         if (c.inventory != null)
         {
             inv_sr.sprite = SpriteManager.current.GetSprite("Inventory", c.inventory.GetName());
-            inv_sr.sortingOrder = c.animation.CurrentSortingOrder + 1;
+            inv_sr.sortingOrder = charSortOrder + 1;
```
Comment (TomMalbran) [inline Assets/Scripts/Controllers/CharacterSpriteController.cs]: Don't add trailing spaces.

**Comment by Raketfart** [comment]
Extra spaces removed now. :/

**Comment by Grenkin1988** [comment]
Conflicts

**Comment by Raketfart** [comment]
Should be fixed now

**Comment by Raketfart** [comment]
I don't know why the heater file is renamed. I've never touched those files. How do I remove that change from my PR?

**Comment by sboigelot** [comment]
@Raketfart it's because it has been renamed from heater->Heater on master and your auto merge reverted it.

Just rename it manually to Heater, it's the simpliest way.

**Comment by Raketfart** [comment]
They were Capitalized in my files. I think I got rid of all the weirdness now

**Pull request merged**